scrapd/core/apd.py

```diff
@@ -286,46 +285,140 @@ def parse_name(name):
         except ValueError:
             pass
         else:
             break
 
-    if dob_index < 0:
-        raise ValueError(f'Cannot parse {Fields.DECEASED}: {deceased_field}')
+    return dob_index
+
+
+def parse_deceased_field(deceased_field):
+    """
+    Parse the deceased field.
+
+    At this point the deceased field, if it exists, is garbage as it contains First Name, Last Name, Ethnicity,
+    Gender, D.O.B. and Notes. We need to explode this data into the appropriate fields.
+
+    :param str deceased_field: the deceased field from the fatality report
+    :return: a dictionary representing a deceased field.
+    :rtype: dict
+    """
+    # Try to parse the deceased fields when the fields are comma separated.
+    try:
+        return parse_comma_delimited_deceased_field(deceased_field)
+    except Exception:
+        pass
+
+    # Try to parse the deceased fields when the fields are pipe separated.
+    try:
+        return parse_pipe_delimited_deceased_field(deceased_field)
+    except Exception:
+        pass
+
+    try:
+        return parse_space_delimited_deceased_field(deceased_field)
+    except Exception:
+        pass
+
+    raise ValueError(f'Cannot parse {Fields.DECEASED}: {deceased_field}')
 
+
+def parse_comma_delimited_deceased_field(deceased_field):
+    """Parse deceased fields seperated with commas.
+
+    :param list split_deceased_field: a list representing the deceased field
+    :return: a dictionary representing the deceased field.
+    :rtype: dict
+    """
     d = {}
-    d[Fields.DOB] = deceased_field[dob_index + 1]
-    notes = deceased_field[dob_index + 2:]
+    split_deceased_field = re.split(r' |(?<=[A-Za-z])/', deceased_field)
+    dob_index = dob_search(split_deceased_field)
+    if dob_index < 0:
+        raise ValueError(f'Cannot find DOB in the deceased field: {deceased_field}')
+    raw_dob = split_deceased_field[dob_index + 1]
+    validated_dob = date_utils.clean_date_string(raw_dob, True)
+    d[Fields.DOB] = validated_dob
+    notes = split_deceased_field[dob_index + 2:]
     if notes:
         d[Fields.NOTES] = ' '.join(notes)
 
     # `fleg` stands for First, Last, Ethnicity, Gender. It represents the info stored before the DOB.
-    fleg = deceased_field[:dob_index]
+    fleg = split_deceased_field[:dob_index]
+    d.update(parse_fleg(fleg))
+    return d
+
+
+def parse_pipe_delimited_deceased_field(deceased_field):
+    """
+    Parse deceased fields separated with pipes.
+
+    :param str deceased_field: the deceased field as a string.
+    :return: a dictionary representing the deceased field.
+    :rtype: dict
+    """
+    d = {}
+    split_deceased_field = deceased_field.split('|')
+    raw_dob = split_deceased_field[-1].strip()
+    d[Fields.DOB] = date_utils.clean_date_string(raw_dob, True)
+
+    fleg = (split_deceased_field[0] + split_deceased_field[1]).split()
+    d.update(parse_fleg(fleg))
+    return d
+
 
+def parse_space_delimited_deceased_field(deceased_field):
```

Comment:
So much delimiter support now 🎉 A lot of similar code between pipe and space, maybe these could be one function in the future with a delimiter parameter.